src/interface/spice/spiceInterface.cpp

```diff
@@ -15,70 +15,74 @@
 #include "tudat/paths.hpp"
 
 #include <math.h>
+#include <mutex>
 
 namespace tudat {
 namespace spice_interface {
 using Eigen::Vector6d;
 
-std::string getCorrectedTargetBodyName(
-        const std::string &targetBodyName )
-{
+static std::mutex spiceMutex;
+
+std::string getCorrectedTargetBodyName(const std::string &targetBodyName) {
     std::string correctedTargetBodyName;
-    if( targetBodyName == "Mercury" ||
-            targetBodyName == "Venus" ||
-            targetBodyName == "MERCURY" ||
-            targetBodyName == "VENUS" ||
-            targetBodyName == "mercury" ||
-            targetBodyName == "venus" )
-    {
+    if (targetBodyName == "Mercury" || targetBodyName == "Venus" ||
```

Comment:
It might be possible to transform the `targetBodyName` string here. Something like this:

```cpp
#include <boost/algorithm/string.hpp>

...


std::string getCorrectedTargetBodyName(const std::string &targetBodyName) {
    std::string correctedTargetBodyName;

    if (boost::algorithm::to_lower(targetBodyName) ==  'mercury' || 
        boost::algorithm::to_lower(targetBodyName) ==  'venus' ) {
        ...
    }
    
    ...
```